java/tools/src/main/java/org/apache/arrow/tools/FileToStream.java

```diff
@@ -38,11 +38,21 @@
   public static void convert(FileInputStream in, OutputStream out) throws IOException {
     BufferAllocator allocator = new RootAllocator(Integer.MAX_VALUE);
     try (ArrowFileReader reader = new ArrowFileReader(in.getChannel(), allocator)) {
-      try (ArrowStreamWriter writer = new ArrowStreamWriter(reader.getVectorSchemaRoot(), reader, out)) {
-        for (ArrowBlock block: reader.getRecordBlocks()) {
-          reader.loadRecordBatch(block);
-          writer.writeBatch();
+      VectorSchemaRoot root = reader.getVectorSchemaRoot();
+      // load the first batch before instantiating the writer so that we have any dictionaries
+      reader.loadNextBatch();
+      try (ArrowStreamWriter writer = new ArrowStreamWriter(root, reader, out)) {
+        writer.start();
+        while (true) {
+          int rowCount = reader.getVectorSchemaRoot().getRowCount();
+          if (rowCount == 0) {
+            break;
+          } else {
+            writer.writeBatch();
+            reader.loadNextBatch();
```

Comment:
I don't think it's a blocker, but having these two classes "wired" together through a mutable row batch (the `root`) seems like a bad code smell. It might be worth eventually changing the API to be

```
reader.loadNextBatch(root)
writer.writeBatch(root)
```